code/core/src/level/elements/Level.java

```diff
@@ -53,6 +56,43 @@ public Level(List<Node> nodes, List<Room> rooms) {
 
         setRandomEnd();
         setRandomStart();
+
+        // Generate tile lookup array while initializing
+        generateTileArray();
+    }
+
+    /**
+     * calculate the global position of the tile and safes them in a two dimensional array for fast
+     * access
+     */
+    private void generateTileArray() {
+        // initalizing levelXOffset, levelYOffset and levelTilesGlobal
+        int minX = Integer.MAX_VALUE;
+        int maxX = Integer.MIN_VALUE;
+        int minY = Integer.MAX_VALUE;
+        int maxY = Integer.MIN_VALUE;
+        for (Room r : getRooms()) {
+            for (Tile[] tA : r.getLayout()) {
+                for (Tile t : tA) {
+                    if (t.getCoordinate().x < minX) minX = t.getCoordinate().x;
```

Comment:
@cagix besteht auf `{` auch für one-line Blöcke. 

Alternativ (und das fände ich schön) 
```suggestion
                    minX = Math.min(minX, t.getCoordinate().x);
```
Für die anderen Aufrufe entsprechend nachziehen.

Reply:
> @cagix besteht auf `{` auch für one-line Blöcke.
> 

hmmm, das ist eigentlich best practice. 

aber: der google java style (und auch aosp) erlauben bei einzeilern mit einem statement tatsächlich den verzicht auf die geschweiften klammern. dann muss die eine anweisung aber tatsächlich auf die selbe zeile passen wie das `if` ...